build.sbt

```diff
@@ -13,6 +13,10 @@ ThisBuild / developers ++= List(
 ThisBuild / githubWorkflowJavaVersions ++=
   Seq(JavaSpec.temurin("17"), JavaSpec(JavaSpec.Distribution.GraalVM("latest"), "17"))
 
+ThisBuild / githubWorkflowBuildConcurrency := Some(
+  JobConcurrency("ci-${{ matrix.java }}-${{ github.ref }}", cancelInProgress = Some(true))
+)
```

Comment:
This is great! Another one that I wonder might make a good default, but I'm not sure.

Reply:
It's relatively easy to block a job by using an incorrect group.
For example, `ci-${{ github.ref }}` does not work with multiple JVMs. And `"ci-${{ matrix.java }}-${{ github.ref }}"` will not work with multiple Scala versions.

On the other hand, if we have complete control of the job matrix, we can generate the correct group:
```scala
val matrix: Map[String, List[String]] = ???
val concurrencyGroup = matrix
  .collect { case (key, values) if values.length > 1 => s"$${{ matrix.$key }}" }
  .mkString("ci-", "-", "")
```

Reply:
Oh hm I see. We do have access to the full matrix in the plugin, so we should be able to do the right thing.

```scala
case (key, values) if values.length > 1 => s"$${{ matrix.$key }}"
```

Is that condition really necessary, is there any harm to include a key even if there is only one value? Because of the "extra jobs" configuration for the matrix there may be multiple values for that key anyway.

Reply:
I enabled concurrency setting by default.